src/xrpld/app/tx/detail/InvariantCheck.h

```diff
@@ -475,6 +473,38 @@ class ValidMPTIssuance
         beast::Journal const&);
 };
 
+/**
+ * @brief Invariants: Permissioned Domains must have some rules and
+ * AcceptedCredentials must have length between 1 and 10 inclusive.
+ *
+ * Since only permissions constitute rules, an empty credentials list
+ * means that there are no rules and the invariant is violated.
```

Comment:
Could you also include that "No duplicates are allowed" and "Input Credentials must be sorted in lexicographic order" -- amongst the list of rules?

I saw unit tests for these invariants, I feel they can be included in the description.